addenda17_test.go

```diff
@@ -102,48 +102,37 @@ func BenchmarkAddenda17String(b *testing.B) {
 func TestValidateAddenda17RecordType(t *testing.T) {
 	addenda17 := mockAddenda17()
 	addenda17.recordType = "63"
-	if err := addenda17.Validate(); err != nil {
-		if e, ok := err.(*FieldError); ok {
-			if e.FieldName != "recordType" {
-				t.Errorf("%T: %s", err, err)
-			}
-		}
+	err := addenda17.Validate()
+	// TODO: are we not expecting any errors here?
```

Comment:
return the fieldError

	if addenda17.recordType != "7" {
		fieldError("recordType", NewErrRecordType(7), addenda17.recordType)
	}